### PR TITLE
Scene Management Pt1

### PR DESCRIPTION
**Pull request by tsboth**

### CODE REVIEW

File: Game/MyProject/Common.xcodeproj/xcuserdata/tamasb.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Bucket
+   type = "1"
```
Comment (krahen94) [inline Game/MyProject/Common.xcodeproj/xcuserdata/tamasb.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:3]: This should not be in the repository, alongside with some other files here. These are local settings that apply only to you.

File: Game/MyProject/Game/SceneManagement/MeshInstance.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+// adds shader uniforms
+void MeshInstance::addShaderUniforms(const std::vector<std::string>& shader_uniforms) {
+    for (std::string i : shader_uniforms)
+        _shader_uniforms.push_back(_shader->GetUniform(i.c_str()));
+}
+
+// sets the world view projection matrix
+void MeshInstance::setWorldViewProjectionMatrix(const IvMatrix44& transform_matrix) {
+    IvRenderer::mRenderer->SetWorldMatrix(transform_matrix);
```
Comment (krahen94) [inline Game/MyProject/Game/SceneManagement/MeshInstance.cpp:62]: Is this 100% correct? Seems to be just a transform matrix but I might be missing out on some details. Even so, this is really confusing.

File: Game/MyProject/Game/SceneManagement/RenderPacket.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    std::shared_ptr<MeshInstance>   _mesh_instance;
+    IvMatrix44                      _world_view_projection_matrix;
+    
+    // protected function(s) and method(s)
+    void setUpDrawing();
+    
+public:
+    // constructor(s) and destructor
+    RenderPacket();
+    RenderPacket(std::shared_ptr<MeshInstance>);
```
Comment (krahen94) [inline Game/MyProject/Game/SceneManagement/RenderPacket.hpp:24]: Generally you want to pass shared pointers as const references

File: Game/MyProject/Game/SceneManagement/Mesh.hpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+class Mesh : std::enable_shared_from_this<Mesh> {
+private:
+    // private variable(s)
+    IvVertexBuffer* _vertex_buffer;
+    IvIndexBuffer*  _index_buffer;
+    
+    // private function(s) and method(s)
+    void destroyVertexBuffer();
+    void destroyIndexBuffer();
+
```
Comment (krahen94) [inline Game/MyProject/Game/SceneManagement/Mesh.hpp:28]: I'd much rather have public stuff before the private one, as that part is the one that people care about when they glance over your header file - the interface that they can use.

**Review by krahen94 — APPROVED** [review]
Looks decent from a first glance., the changes I am requesting are minor and not absolutely necessary.

**Comment by raxvan** [comment]
RenderPacket needs some bigger changes. It must not own any member variable, no virtual functions.

Mesh,MeshInstance, requires some smaller changes. const& getters, and improved mesh creation.

**Pull request closed without merging**